builderror

/home/travis/build/miyabayt/spring-boot-doma2-sample/sample-web-base/src/main/java/com/sample/web/base/aop/SetModelAndViewInterceptor.java:70: error: Cannot use 'val' here because initializer expression does not have a representable type: Type cannot be resolved
        val codeCategories = codeCategoryService.fetchAll();
            ^
1 error
1 warning
:sample-web-base:compileJava FAILED

--- a/sample-domain/src/main/java/com/sample/domain/dao/system/CodeCategoryDao.java
+++ b/sample-domain/src/main/java/com/sample/domain/dao/system/CodeCategoryDao.java
@@ -27,6 +27,14 @@ public interface CodeCategoryDao {
             final Collector<CodeCategory, ?, R> collector);
 
     /**
+     * コード分類定義を全件取得します。
+     *
+     * @return
+     */
+    @Select
+    List<CodeCategory> fetchAll();
+
+    /**
      * コード分類を1件取得します。
      *
      * @param id

--- a/sample-domain/src/main/java/com/sample/domain/service/system/CodeCategoryService.java
+++ b/sample-domain/src/main/java/com/sample/domain/service/system/CodeCategoryService.java
@@ -17,6 +17,8 @@ import com.sample.domain.service.BaseTransactionalService;
 
 import lombok.val;
 
+import java.util.List;
+
 /**
  * コード分類サービス
  */
@@ -40,6 +42,17 @@ public class CodeCategoryService extends BaseTransactionalService {
         val codeCategories = codeCategoryDao.selectAll(where, options, toList());
 
         return pageFactory.create(codeCategories, pageable, options.getCount());
+    }
+
+    /**
+     * コード分類を全件取得します。
+     *
+     * @return
+     */
+    @Transactional(readOnly = true)
+    public List<CodeCategory> fetchAll() {
+        val codeCategories = codeCategoryDao.fetchAll();
+        return codeCategories;
     }
 
     /**

--- a/sample-domain/src/main/resources/META-INF/com/sample/domain/dao/system/CodeCategoryDao/fetchAll.sql
+++ b/sample-domain/src/main/resources/META-INF/com/sample/domain/dao/system/CodeCategoryDao/fetchAll.sql
@@ -1,0 +1,8 @@
+SELECT
+    /*%expand*/*
+FROM
+    code_category
+WHERE
+    deleted_at IS NULL
+ORDER BY
+    1

--- a/sample-web-base/src/main/java/com/sample/web/base/aop/SetModelAndViewInterceptor.java
+++ b/sample-web-base/src/main/java/com/sample/web/base/aop/SetModelAndViewInterceptor.java
@@ -64,16 +64,9 @@ public class SetModelAndViewInterceptor extends BaseHandlerInterceptor {
      * @return
      */
     protected List<CodeCategory> getCodeCategories() {
-        List<CodeCategory> data = Collections.emptyList();
-
         // コード分類をすべて取得する
         val codeCategories = codeCategoryService.fetchAll();
-
-        if (codeCategories != null) {
-            data = codeCategories.getData();
-        }
-
-        return data;
+        return codeCategories;
     }
 
     /**

